generator/tree.go

```diff
@@ -200,11 +200,54 @@ func getMetricNode(oid string, node *Node, nameToNode map[string]*Node) (*Node,
 	return n, oidInstance
 }
 
+func copyTree(node *Node) *Node {
+	newNode := *node
+	newNode.Children = make([]*Node, 0, len(node.Children))
+	newNode.Indexes = make([]string, len(node.Indexes))
+	copy(newNode.Indexes, node.Indexes)
+	// Deep copy children.
+	for _, child := range node.Children {
+		newNode.Children = append(newNode.Children, copyTree(child))
+	}
+	return &newNode
+}
+
 func generateConfigModule(cfg *ModuleConfig, node *Node, nameToNode map[string]*Node) *config.Module {
 	out := &config.Module{}
 	needToWalk := map[string]struct{}{}
 	tableInstances := map[string][]string{}
 
+	// Gather all type overrides for the current module.
```

Comment:
I think this can be done in one pass

Reply:
By gathering first we can make sure there are type overrides in the configuration and only duplicate the tree when actually needed.

Reply:
It is possible to do in one pass and do the duplication lazily, flow will suffer a bit though.